spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -27,12 +27,12 @@ module.exports = function (provider, setup) {
         console.warn('written');
         written = true;
       });
-      setTimeout(function () {
+      dispatch.gotMessageAsync('onData', [], function(msg) {
+        console.warn('received onData');
         expect(written).toBe(true);
         expect(dispatch.gotMessage('onData', [])).not.toEqual(false);
-        done();
-        socket.close(function () {});
-      }, 500);
+        socket.close(function () { done(); });
```

Comment:
This can be written slightly more elegantly as `socket.close(done)` (eta-contraction!)